test/proxy.test.js

```diff
@@ -1,44 +1,50 @@
 var constants = require('../config/constants');
 var RdGlobalConfig = constants.RdGlobalConfig;
 var nock = require('nock');
-var RdHandler = require('../src/server');
+var RdHandler = require('../src/proxy');
 var http = require('http');
 var assert = require('chai').assert;
 var testHelper = require('./testHelper');
 
+
 describe('RdHandler', function () {
   context('Proxy Server', function () {
-
+    var originalScheme;
     before(function (done) {
       this.timeout = 5000;
-      testHelper.nockGetRequest(constants.HUB_STATUS_URL, 'http', null, 200);
       testHelper.initializeDummyLoggers();
       testHelper.initializeDummyHandlers();
+      testHelper.initializeDummyProxy();
+      originalScheme = RdGlobalConfig.SCHEME;
+      RdGlobalConfig.SCHEME = 'http';
       
-      RdHandler.startProxy(RdGlobalConfig.RD_HANDLER_PORT, function (port) {
-        console.log('Test Network Utility Proxy Started on Port: ', port);
+      RdHandler.startProxyServer(RdGlobalConfig.RD_HANDLER_PROXY_PORT, function (port) {
+        console.log('Test Network Utility Proxy Server Started on Port: ', port);
         done();
       });
     });
 
     after(function (done) {
       this.timeout = 5000;
-      RdHandler.stopProxy(function () {
+      RdHandler.stopProxyServer(function () {
         done();
       });
       testHelper.deleteLoggers();
       testHelper.deleteHandlers();
       nock.cleanAll();
+      RdGlobalConfig.SCHEME.restore();
+      RdGlobalConfig.SCHEME = originalScheme;
     });
 
     it('Requests on behalf of the client and returns the response', function (done) {
       this.timeout = 5000;
+      testHelper.nockGetRequest(constants.HUB_STATUS_URL, 'http', null, 200);
       var reqOptions = {
         method: 'GET',
-        host: 'localhost',
-        port: RdGlobalConfig.RD_HANDLER_PORT,
+        host: constants.HUB_HOST,
```

Comment:
shouldn't this be `localhost`? Since it's not reverse proxy here. We are making a request on the `proxy` server.

This request is going to the hub directly and not our proxy server...

Reply:
done

Reply:
can be resolved...